modules/simple_forms_api/app/services/simple_forms_api/notification_email.rb

```diff
@@ -1,26 +1,31 @@
 # frozen_string_literal: true
 
 module SimpleFormsApi
-  class ConfirmationEmail
-    attr_reader :form_number, :confirmation_number, :user
+  class NotificationEmail
+    attr_reader :form_number, :confirmation_number, :notification_type, :user
 
     TEMPLATE_IDS = {
-      'vba_21_0845' => Settings.vanotify.services.va_gov.template_id.form21_0845_confirmation_email,
-      'vba_21p_0847' => Settings.vanotify.services.va_gov.template_id.form21p_0847_confirmation_email,
-      'vba_21_0966' => Settings.vanotify.services.va_gov.template_id.form21_0966_confirmation_email,
-      'vba_21_0972' => Settings.vanotify.services.va_gov.template_id.form21_0972_confirmation_email,
-      'vba_21_4142' => Settings.vanotify.services.va_gov.template_id.form21_4142_confirmation_email,
-      'vba_21_10210' => Settings.vanotify.services.va_gov.template_id.form21_10210_confirmation_email,
-      'vba_20_10206' => Settings.vanotify.services.va_gov.template_id.form20_10206_confirmation_email,
-      'vba_20_10207' => Settings.vanotify.services.va_gov.template_id.form20_10207_confirmation_email,
-      'vba_40_0247' => Settings.vanotify.services.va_gov.template_id.form40_0247_confirmation_email
+      'vba_21_0845' => { confirmation: Settings.vanotify.services.va_gov.template_id.form21_0845_confirmation_email },
+      'vba_21p_0847' => { confirmation: Settings.vanotify.services.va_gov.template_id.form21p_0847_confirmation_email },
+      'vba_21_0966' => { confirmation: Settings.vanotify.services.va_gov.template_id.form21_0966_confirmation_email },
+      'vba_21_0972' => { confirmation: Settings.vanotify.services.va_gov.template_id.form21_0972_confirmation_email },
+      'vba_21_4142' => { confirmation: Settings.vanotify.services.va_gov.template_id.form21_4142_confirmation_email },
+      'vba_21_10210' => {
+        confirmation: Settings.vanotify.services.va_gov.template_id.form21_10210_confirmation_email,
+        error: Settings.vanotify.services.va_gov.template_id.form21_10210_error_email,
+        received: Settings.vanotify.services.va_gov.template_id.form21_10210_received_email
+      },
+      'vba_20_10206' => { confirmation: Settings.vanotify.services.va_gov.template_id.form20_10206_confirmation_email },
+      'vba_20_10207' => { confirmation: Settings.vanotify.services.va_gov.template_id.form20_10207_confirmation_email },
+      'vba_40_0247' => { confirmation: Settings.vanotify.services.va_gov.template_id.form40_0247_confirmation_email }
     }.freeze
     SUPPORTED_FORMS = TEMPLATE_IDS.keys
```

Comment:
My mind can't help but go to dynamically generating this list. I'd love your thoughts on something like this?

```ruby
TEMPLATE_IDS = build_template_ids.freeze
SUPPORTED_FORMS = TEMPLATE_IDS.keys

...

private

def build_template_ids
  form_templates = {
    'vba_21_0845' => %i[confirmation],
    'vba_21p_0847' => %i[confirmation],
    'vba_21_0966' => %i[confirmation],
    'vba_21_0972' => %i[confirmation],
    'vba_21_4142' => %i[confirmation],
    'vba_21_10210' => %i[confirmation error received],
    'vba_20_10206' => %i[confirmation],
    'vba_20_10207' => %i[confirmation],
    'vba_40_0247' => %i[confirmation]
  }

  form_templates.each_with_object({}) do |(form, template_types), hash|
    hash[form] = template_types.each_with_object({}) do |type, inner_hash|
      inner_hash[type] = Settings.vanotify.services.va_gov.template_id.public_send("#{form}_#{type}_email")
    end
  end
end
```

Reply:
Actually, I've decided I'm getting too "fancy" with it. Disregard 😂

Reply:
👀 ooo that's interesting! I think it's a little much for this configuration hash. I think there's benefit to being able to see it all clearly (even if verbosely). But its a nice idea!